### ISSUE TITLE
Support OS notifications 

### ISSUE THREAD

**Issue by Schroedingers-Cat** [garethgeorge/backrest#585]
**Is your feature request related to a problem? Please describe.**
If scheduled backups fail silently in the background, users might not notice (for instance, they're not looking into the log files) and are without backups for days or weeks.

**Describe the solution you'd like**
Every OS these days has support for notifications and it would be great if backrest would add support for that. The feature list mentions notification support for third-party services like Discord or Slack, but a local notification would be useful as well, esp. as default value for a new configuration. 

Notifications could be sent when a backup finishes, fails or misses a variable amount of runs.

**Additional context**
I'm currently using custom scripts to create those notifications on Linux and Windows but that is not a viable solution to a lot of users.

**Comment by garethgeorge** [comment]
Looks like this is something backrest could integrate support for with https://github.com/gen2brain/beeep .

Hook types are relatively straightforward to add to the hook registry https://github.com/garethgeorge/backrest/tree/main/internal/hook/types though some amount of plumbing is needed for UI support.

**Comment by Schroedingers-Cat** [comment]
> scripts are a solid workaround  

Yes, scripting gets the basics done (notification per executed backup job).  

Looking at how backup notifications are done on Windows 10  or on macOS, I think those solutions are, however, more reliable than scripts. For instance, backups might start only if the computer is on power supply or if it can reach a specific host or network. Now, should those conditions never be met for instance for two weeks or a month, Windows and macOS will create a notification to make users aware about this. Another example is a monthly backup job I scheduled using resticprofile including notifications on errors, but that job didn't execute for months so none of my notification scripts got actually triggered.  

On both cases, you'd be receive a warning on Windows and macOS.

**Comment by johan-scriptdrift** [comment]
pr https://github.com/garethgeorge/backrest/pull/679

**Comment by tredondo** [comment]
How about [browser notifications](https://developer.mozilla.org/en-US/docs/Web/API/Notifications_API) in the interim? They're supported from localhost, which is considered a [secure origin](https://developer.mozilla.org/en-US/docs/Web/Security/Defenses/Secure_Contexts#potentially_trustworthy_origins).
